test/Playground.js

```diff
@@ -43,8 +46,30 @@ describe("Playground Contract", () => {
     }
   };
 
+  const niceDeposit = async function(amount) {
+    await playground.connect(shareholders[0].signer).
+    setApprovalForAll(playground.address, true);
```

Comment:
`setApprovalForAll` seems risky. I tried `approve(playground.address, amount)` but got `Error: VM Exception while processing transaction: reverted with reason string 'ERC721: owner query for nonexistent token'`. I could not figure out why yet.